src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -205,35 +231,81 @@ private OAuthRealm(
         {
             assert nextRealmBit < MAX_REALMS;
             this.realmName = realmName;
-            this.realmId = 1L << nextRealmBit++ << MAX_SCOPES;
         }
 
         private long resolve(
-            String[] scopeNames)
+            String[] scopeNames,
+            String issuerName,
+            String audienceName)
         {
             long authorization = NO_AUTHORIZATION;
-            if(nextScopeBit + scopeNames.length < MAX_SCOPES)
+            if (nextScopeBit + scopeNames.length < MAX_SCOPES)
             {
-                authorization = realmId;
+                final OAuthRealmInfo realmInfo = computeInfoIfAbsent(issuerName, audienceName);
+                authorization = realmInfo.realmId;
                 for (int i = 0; i < scopeNames.length; i++)
                 {
-                    authorization |= scopeBitsByName.computeIfAbsent(scopeNames[i], this::assignScopeBit);
+                    authorization |= realmInfo.computeScopeBitsIfAbsent(scopeNames[i], this::assignScopeBit);
                 }
             }
             return authorization;
         }
 
         private long lookup(
-            String[] scopeNames)
+            String[] scopeNames,
+            String issuerName,
+            String audienceName)
```

Comment:
Recommend refactoring the order of these parameters to `issuerName`, `audienceName`, `scopeNames`.